### PR TITLE
chore: Update v12.5.0 with master

### PR DESCRIPTION
**Pull request by hjetpoluru**
## **Description**

Merged latest master to v12.5.0

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/27797?quickstart=1)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,108 +9,37 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [12.5.0]
 
 ## [12.4.0]
```
Comment (danjm) [inline CHANGELOG.md:11]: This first `## [12.4.0]` should not be here

File: CHANGELOG.md [hunk 2]
```diff
@@ -5582,6 +5149,8 @@ Update styles and spacing on the critical error page  ([#20350](https://github.c
 
 [Unreleased]: https://github.com/MetaMask/metamask-extension/compare/v12.5.0...HEAD
 [12.5.0]: https://github.com/MetaMask/metamask-extension/compare/v12.4.0...v12.5.0
+[Unreleased]: https://github.com/MetaMask/metamask-extension/compare/v12.4.1...HEAD
```
Comment (danjm) [inline CHANGELOG.md:5152]: Also, there is a changelog error here

File: CHANGELOG.md (hunk 1 above)
Comment (hjetpoluru) [reply CHANGELOG.md]: Thanks @danjm, now its addressed after your comment.

File: CHANGELOG.md (hunk 2 above)
Comment (hjetpoluru) [reply CHANGELOG.md]: I have noted the instruction you have provided in the meeting, especially regarding resolving the changelog by running the command `yarn lint:changelog` and manually checking the file.

**Pull request merged**